Api/Src/Routes/TeamProjectRoute.ts

```diff
@@ -1,14 +1,16 @@
 import * as express from 'express';
 import TeamProjectController from '../Controllers/TeamProjectController';
+import { HasRole } from '../Middlewares/HasRole';
+import { UserType } from '../Models/User';
 
 const courseRoute = (controller: TeamProjectController) => {
     return (router: express.Router) => {
-        router.route("/teams/projects").get(controller.getTeamProjects);
-        router.route("/teams/projects/:id").get(controller.getTeamProjectById);
-        router.route("/teams/:id/projects").get(controller.getTeamProjectsByTeamId);
-        router.route("/teams/projects").post(controller.createTeamProject);
-        router.route("/teams/projects/:id").put(controller.updateTeamProject);
-        router.route("/teams/projects/:id").delete(controller.deleteTeamProject);
+        router.route("/teams/projects").get(HasRole([UserType.Admin]), controller.getTeamProjects);
+        router.route("/teams/projects/:id").get(HasRole([UserType.Admin]), controller.getTeamProjectById);
+        router.route("/teams/projects").post(HasRole([UserType.Admin]), controller.createTeamProject);
+        router.route("/teams/projects/:id").put(HasRole([UserType.Admin]), controller.updateTeamProject);
+        router.route("/teams/projects/:id").delete(HasRole([UserType.Admin]), controller.deleteTeamProject);
```

Comment:
Good point 👍 

In my opinion mentors should not have access to this endpoints because,
mentors would be allowed to get/create/modify/delete projects for other teams.

From the other side, mentors should have access to get/create/modify/delete their own projects.

I will solve it by create new endpoints for these purposes.

Reply:
Maybe we should talk about it with other members of our team? I don't know if it has sense to do new endpoints, when at monday we should end our project.